CONTRIBUTING.md

```diff
@@ -0,0 +1,133 @@
+# Contributing to `linter`
+
+:+1::tada: First off, thanks for taking the time to contribute! :tada::+1:
+
+The following is a set of guidelines for contributing to linter.
+These are just guidelines, not rules, use your best judgment and feel free to
+propose changes to this document in a pull request.
+
+This project adheres to the [Open Code of Conduct][code-of-conduct]. By participating, you are expected to uphold this code.
+[code-of-conduct]: http://todogroup.org/opencodeofconduct/#Atom/opensource@github.com
+
+## Submitting Issues
+
+* You can create an issue [here](https://github.com/atom-community/linter/issues/new), but
+  before doing that please read the notes below on collecting information and submitting issues.
+  Include as many details as possible with your report.
+* Include the version of Atom you are using and the OS.
+* Include screenshots and animated GIFs whenever possible; they are immensely
+  helpful.
+* Include the behavior you expected and other places you've seen that behavior
+  such as Emacs, vi, Xcode, etc.
+* Check the dev tools (`alt-cmd-i`) for errors to include. If the dev tools
+  are open _before_ the error is triggered, a full stack trace for the error
+  will be logged. If you can reproduce the error, use this approach to get the
+  full stack trace and include it in the issue.
+* On Mac, check Console.app for stack traces to include if reporting a crash.
+* Perform a [cursory search](https://github.com/atom-community/linter/search?q=&type=Issues&utf8=%E2%9C%93)
+  to see if a similar issue has already been submitted.
+* Please setup a [profile picture](https://help.github.com/articles/how-do-i-set-up-my-profile-picture)
+  to make yourself recognizable and so we can all get to know each other better.
+
+## Pull Requests
+
+* We prefer small, focused, single-responsibility pull requests that include tests where possible. These can be contrasted with large pull requests, pull requests with multiple unrelated concerns, and pull requests which have no tests.
+* Include screenshots and animated GIFs in your pull request whenever possible.
+* **Please ensure that your pull request has no lint errors.** This is a project for linters after all,
+  so please ensure you have the [linter-coffeelint](https://atom.io/packages/linter-coffeelint) and [linter-eslint](https://atom.io/packages/linter-eslint) packages installed in
+  Atom.
+* Include thoughtfully-worded, well-structured
+  [Jasmine](http://jasmine.github.io/) specs in the `./spec` folder. Run them using `apm test`. See
+  the [Specs Styleguide](#specs-styleguide) below.
+* Document new code based on the
+  [Documentation Styleguide](#documentation-styleguide)
+* End files with a newline.
```

Comment:
It might be worth noting that both Atom and the GitHub Online Editor do this for you.